x-pack/plugins/profiling/public/components/stacked_bar_chart.tsx

```diff
@@ -5,47 +5,109 @@
+  Chart,
+  HistogramBarSeries,
+  Settings,
+  StackMode,
+  timeFormatter,
+  TooltipInfo,
+  XYChartElementEvent,
+} from '@elastic/charts';
+import { EuiPanel } from '@elastic/eui';
 import React, { useContext } from 'react';
-
-import { Axis, BarSeries, Chart, Settings } from '@elastic/charts';
-
-import { timeFormatter } from '@elastic/charts';
-
+import { TopNSample, TopNSubchart } from '../../common/topn';
+import { chartTheme } from '../utils/chart_styles';
+import { asPercentage } from '../utils/formatters/as_percentage';
 import { TopNContext } from './contexts/topn';
+import { SubChart } from './subchart';
+
+function SubchartTooltip({
+  highlightedSubchart,
+}: TooltipInfo & { highlightedSubchart: TopNSubchart }) {
+  return (
+    <EuiPanel>
+      <SubChart
+        index={highlightedSubchart.Index}
+        color={highlightedSubchart.Color}
+        category={highlightedSubchart.Category}
+        percentage={highlightedSubchart.Percentage}
+        data={highlightedSubchart.Series}
+        height={200}
+        width={400}
+      />
+    </EuiPanel>
+  );
+}
 
 export interface StackedBarChartProps {
-  id: string;
-  name: string;
   height: number;
-  x: string;
-  y: string;
-  category: string;
+  asPercentages: boolean;
+  onBrushEnd: (range: { rangeFrom: string; rangeTo: string }) => void;
+  onSampleClick: (sample: TopNSample) => void;
+  onSampleOut: () => void;
+  highlightedSubchart?: TopNSubchart;
 }
 
 export const StackedBarChart: React.FC<StackedBarChartProps> = ({
-  id,
-  name,
   height,
-  x,
-  y,
-  category,
+  asPercentages,
+  onBrushEnd,
+  onSampleClick,
+  onSampleOut,
+  highlightedSubchart,
 }) => {
   const ctx = useContext(TopNContext);
 
   return (
     <Chart size={{ height }}>
-      <Settings showLegend={false} />
-      <BarSeries
-        id={id}
-        name={name}
-        data={ctx.samples}
-        xAccessor={x}
-        yAccessors={[y]}
-        stackAccessors={[x]}
-        splitSeriesAccessors={[category]}
+      <Settings
+        showLegend={false}
+        tooltip={{
+          customTooltip: highlightedSubchart
+            ? (props) => <SubchartTooltip {...props} highlightedSubchart={highlightedSubchart} />
+            : () => <></>,
+        }}
+        brushAxis={BrushAxis.X}
+        onBrushEnd={(brushEvent) => {
+          const rangeFrom = new Date(brushEvent.x![0]).toISOString();
+          const rangeTo = new Date(brushEvent.x![1]).toISOString();
+
+          onBrushEnd({ rangeFrom, rangeTo });
+        }}
+        theme={chartTheme}
+        onElementClick={(events) => {
+          const [value] = events[0] as XYChartElementEvent;
+          onSampleClick(value.datum as TopNSample);
+        }}
+        onElementOver={() => {
+          onSampleOut();
+        }}
+        onElementOut={() => {
+          onSampleOut();
+        }}
       />
+      {ctx.charts.map((chart) => (
+        <HistogramBarSeries
+          key={chart.Category}
+          id={chart.Category}
```

Comment:
I've also noticed that the `chart.Category` can be a number type as in the sample dataset you provided me.
This unfortunately is not captured by your typing (probably due to a type assertion or some any in the fetching mechanism) and it flows down the code up to the chart. 
This for example is the response from the topN call where you can see that `Category` is a number not a string as requested by the `TopNSample` type:
```ts
export interface CountPerTime {
  Timestamp: number;
  Count: number | null;
}

export interface TopNSample extends CountPerTime {
  Category: string;
}
```
<img width="642" alt="Screenshot 2022-07-19 at 13 01 29" src="https://user-images.githubusercontent.com/1421091/179735327-5cc64c12-b06e-4c68-b987-87296549eefd.png">


This makes the chart breaks (it breaks for me on mouse hover)

Reply:
I'm curious about the conditions that cause the chart to break on hover but also how the `hosts` endpoint is returning a `Category` as a number. Unfortunately, I couldn't replicate it locally.

For all of the stack trace endpoints (`container`, `hosts`, etc), we should expect `Category` to be an opaque string. First. the histogram bucket keys are strings as returned by Elasticsearch. Second, when we generate the response on the Kibana server, we use `createTopNSamples` to group the categories, which are typed as strings.

In particular for the hosts endpoint, the category should be a string because it represents a 64-bit host ID. If we sent it as a number in the response, we would lose precision when assigning it to a number, thus, changing the host ID.

If we need to fix the types, my suggestion would be to fix it in the server route so that we can maintain the guarantee that categories are always strings to prevent the above precision issue.

Reply:
I can't reproduce this either. Maybe it's an `elasticdump` artifact. @markov00 do you have a reproduction scenario?